sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java

```diff
@@ -2320,6 +2322,46 @@ public void testWriteToTableDecorator() throws Exception {
     p.run();
   }
 
+  @Test
+  public void testWriteWithAllowTruncatedTimestamps() throws IOException, InterruptedException {
+    if (useStorageApi) {
+      // TODO: to support storage API, changes have to be made to
+      // org.apache.beam.sdk.io.gcp.bigquery.BeamRowToStorageApiProto
+      return;
```

Comment:
Could this test that we fail gracefully in this case? It would also be good to file a jira for the follow up work and reference that, e.g. TODO(BEAM-XXX):

Reply:
@TheNeuralBit I'm going to create the Jira ticket. What do you mean by "fail gracefully"? Can you elaborate a little more, thanks.